gsa/src/web/pages/extras/trashcanpage.js

```diff
@@ -226,344 +202,312 @@ class Trashcan extends React.Component {
+        trash.config_list,
+      );
+      const renderConfigs = isDefined(trash.config_list);
+      const renderPolicies = isDefined(trash.config_list);
+
+      return (
+        <TableBody>
+          {render_alerts &&
+            createContentRow('alert', _('Alerts'), trash.alert_list.length)}
+          {renderAudits &&
+            createContentRow('audit', _('Audits'), audits.length)}
+          {render_credentials &&
+            createContentRow(
+              'credential',
+              _('Credentials'),
+              trash.credential_list.length,
+            )}
+          {render_filters &&
+            createContentRow('filter', _('Filters'), trash.filter_list.length)}
+          {render_groups &&
+            createContentRow('group', _('Groups'), trash.group_list.length)}
+          {render_notes &&
+            createContentRow('note', _('Notes'), trash.note_list.length)}
+          {render_overrides &&
+            createContentRow(
+              'override',
+              _('Overrides'),
+              trash.override_list.length,
+            )}
+          {render_permissions &&
+            createContentRow(
+              'permission',
+              _('Permissions'),
+              trash.permission_list.length,
+            )}
+          {renderPolicies &&
+            createContentRow('policy', _('Policies'), policies.length)}
+          {render_port_lists &&
+            createContentRow(
+              'port_list',
+              _('Port Lists'),
+              trash.port_list_list.length,
+            )}
+          {render_report_formats &&
+            createContentRow(
+              'report_format',
+              _('Report Formats'),
+              trash.report_format_list.length,
+            )}
+          {render_roles &&
+            createContentRow('role', _('Roles'), trash.role_list.length)}
+          {renderConfigs &&
+            createContentRow('config', _('Scan Configs'), configs.length)}
+          {render_scanners &&
+            createContentRow(
+              'scanner',
+              _('Scanners'),
+              trash.scanner_list.length,
+            )}
+          {render_schedules &&
+            createContentRow(
+              'schedule',
+              _('Schedules'),
+              trash.schedule_list.length,
+            )}
+          {render_tags &&
+            createContentRow('tag', _('Tags'), trash.tag_list.length)}
+          {render_targets &&
+            createContentRow('target', _('Targets'), trash.target_list.length)}
+          {renderTasks && createContentRow('task', _('Tasks'), tasks.length)}
+          {render_tickets &&
+            createContentRow('ticket', _('Tickets'), trash.ticket_list.length)}
+        </TableBody>
+      );
+    },
+    [createContentRow],
+  );
+
+  if (!isDefined(trash) && !isDefined(error)) {
+    return <Loading />;
+  } else if (!isDefined(trash) && isDefined(error)) {
+    setTrash({});
```

Comment:
I don't think this works. You are changing the state during a react render phase.

Reply:
I have set it to watch for `loading` instead. trash should always be defined since it's initialized as `{}` and never set to undefined in code.